src/componentWithName.js

```diff
@@ -16,7 +16,7 @@ function hasName(name, prop, propName, componentName) {
   }
 
   const { type } = prop;
-  if (getComponentName(type) !== name) {
+  if (!new RegExp(name).test(getComponentName(type))) {
```

Comment:
Well to be fair, doing `getComponentName(type).match(name)` also worked without creating a new regex but is way less performant overall.

Before that, I also had `getComponentName(type) !== name && new RegExp(name).test(getComponentName(type)` which would only create a RegExp if the strings were not directly equal.

I could also do a third conditionally specifically to check `match` or `test`... but I do kind of like the idea of being able to pass `/foo$/` directly instead of passing `new RegExp(/foo$/)` although I guess the latter is the most explicit. Is forcing the user to creating a RegExp and then checking for a `RegExp` instance specifically what you would prefer @ljharb?